src/EventSubscriber/DomainRedirectEventSubscriber.php

```diff
@@ -34,12 +64,38 @@ static function getSubscribedEvents() {
   /**
    * This method is called whenever the kernel.request event is
    * dispatched.
-   * @todo remove hardcoded URL and create a configuration form to manage redirection
+   *
+   * @todo Needs a service which will handle the exclusion/inclusion of
+   * the mentioned path/page.
+   *
    * @param GetResponseEvent $event
    */
   public function requestHandler(GetResponseEvent $event) {
-     $response = new RedirectResponse("http://www.google.com",301);
-     $response->send();
+    // If user has 'bypass' permission, then no need to process further.
+    if ($this->userAccount->hasPermission('bypass domain 301 redirect')) {
+      return;
+    }
+
+    $domain_config = $this->configFactory->get('domain_301_redirect.settings')->getRawData();
+    // If domain redirection is not enabled, then no need to process further.
+    if (!$domain_config['domain_301_redirect_enabled']) {
+      return;
+    }
+
+    if (!preg_match('|^https?://|', $domain_config['domain_301_redirect_domain'])) {
```

Comment:
@joshirohit100  Thanks for the work. All seems good to me. Just one input- If you can  briefly write the logic , it would be great help to others working and using the module.
Thanks again.